src/Fcf/Data.hs

```diff
@@ -129,6 +136,17 @@ data Head :: [a] -> Exp (Maybe a)
 type instance Eval (Head '[]) = 'Nothing
 type instance Eval (Head (a ': _as)) = 'Just a
 
+data Last :: [a] -> Exp (Maybe a)
+type instance Eval (Last '[]) = 'Nothing
+type instance Eval (Last (a ': as)) =
+  Eval (If (Eval (Null as)) (Pure ('Just a)) (Last as))
```

Comment:
You could also do a similar case split as for `Init` here.

Reply:
Thanks for your quick feedbacks!